components/AposWidget.astro

```diff
@@ -37,19 +41,22 @@ if (isEdit) {
 } else if (shouldWrapWithAnchor) {
   WrapperElement = 'div';
   wrapperAttributes[anchorAttribute] = widget.anchorId;
+} else {
+  WrapperElement = Fragment;
 }
 
-const shouldWrap = isEdit || shouldWrapWithAnchor;
----
+if (options?.aposStylesWrapper === false) {
+  StylesWrapperElement = Fragment;
+  stylesElements = null;
+  stylesAttributes = {};
+}
 
-{ Component &&
-  <>
-    {shouldWrap ? (
-      <WrapperElement {...wrapperAttributes}>
-        <Component {widget} {options} {...props}/>
-      </WrapperElement>
-    ) : (
+---
+{Component &&
+  <WrapperElement {...wrapperAttributes}>
```

Comment:
I'm not entirely sure we should have multiple wrappers here. The outer one comes usually from the anchors module? Why not combine them (where the styles attributes override the others to avoid strange things happen). 

I think layout module also uses such technique, but it opt-outs of auto wrapper anyways. You should have a different task regarding that in Astro.